src/datapool/datapool_pmem.c

```diff
@@ -153,13 +163,18 @@ datapool_flag_clear(struct datapool *pool, int flag)
  * finish successfully.
  */
 struct datapool *
-datapool_open(const char *path, size_t size, int *fresh, bool prefault)
+datapool_open(const char *path, const char* user_signature, size_t size, int *fresh, bool prefault)
 {
     struct datapool *pool = cc_alloc(sizeof(*pool));
     if (pool == NULL) {
         log_error("unable to create allocate memory for pmem mapping");
         goto err_alloc;
     }
+    if (!user_signature ||
```

Comment:
write why.

Reply:
I added a comment.
Regarding first checking if user_signature is null pointer is to avoid calling strnlen with null pointer, which would be undefined behavior.